File: LoginUser.c
#include <stdio.h>
#include <string.h>

// Function to simulate checking user credentials in the database
int checkLogin(const char *email, const char *password) {
    if (strcmp(email, "user@example.com") == 0 && strcmp(password, "password123") == 0) {
        return 1;
    }
    return 0;
}

// Procedure to perform logout
void doLogout() {
    printf("You have been logged out.\n");
}

int main() {
    char email[50];
    char password[50];
    int isUserLoggedIn = 0;

    printf("Enter your email: ");
    scanf("%s", email);
    printf("Enter your password: ");
    scanf("%s", password);

    if (checkLogin(email, password)) {
        isUserLoggedIn = 1;
        printf("Login successful! Welcome, %s\n", email);
    } else {
        printf("Login failed. Please try again.\n");
    }

    // Simulate logout
    if (isUserLoggedIn) {
        printf("Do you want to logout? (yes/no): ");
        scanf("%s", password);
        if (strcmp(password, "yes") == 0) {
            doLogout();
        }
    }

    return 0;
}

File: LoginUser.pas
program LoginUser;

uses
  SysUtils;

var
  email, password: string;
  isUserLoggedIn: boolean;

// Function to simulate checking user credentials in the database
function checkLogin(email, password: string): boolean;
begin
  if (email = 'user@example.com') and (password = 'password123') then
    Result := true
  else
    Result := false;
end;

// Procedure to perform logout
procedure doLogout;
begin
  isUserLoggedIn := false;
  Writeln('You have been logged out.');
end;

begin
  isUserLoggedIn := false;
  Writeln('Enter your email: ');
  Readln(email);
  Writeln('Enter your password: ');
  Readln(password);

  if checkLogin(email, password) then
  begin
    isUserLoggedIn := true;
    Writeln('Login successful! Welcome, ', email);
  end
  else
    Writeln('Login failed. Please try again.');

  // Simulate logout
  Writeln('Do you want to logout? (yes/no)');
  Readln(password);
  if password = 'yes' then
    doLogout;
end.
